### PR TITLE
docs: add dark mode

### PR DESCRIPTION
**Pull request by laurenceisla**
Adds dark mode to the documentation using [`sphinx-rtd-dark-mode`](https://pypi.org/project/sphinx-rtd-dark-mode/#description).

### CODE REVIEW

**Comment by laurenceisla** [comment]
Testing to see how it looks. The code blocks may look a bit too clear cojpared to the rest of the page.

**Comment by steve-chavez** [comment]
Looks cool!

> The code blocks may look a bit too clear cojpared to the rest of the page.

Yeah, everything looks good except the snippets

**Comment by laurenceisla** [comment]
> The different logos on the index page look a bit inconsistent as well...

Yeah, I modified those with white background and added a border to the letters. Looks better, but not sure if it should be kept this way.

There are also problems with the generated images, e.g. in tutorial 0 there are black arrows and letters that are not readable at all. A white border for those elements or a white background for the whole image (easier) is needed.

**Comment by steve-chavez** [comment]
> Yeah, I modified the logos that have white background and added a white border only to the letters. Looks better, but not sure what you think. It may be modified in another PR too.

I like it. And yeah, things can be improved on another PR.

**Review by steve-chavez — APPROVED** [review]

**Comment by laurenceisla** [comment]
I was hoping to not change the logo itself by adding a bit of background, but yeah it changes the silhouette too much. 

> Tembo, too, from first glance

 IIRC that logo ls meant for both dark/light mode.

But yes, the sponsors have the last saying in this, I'll notify them to provide a dark-mode logo. Since it would take some time I'll remove that commit and merge as-is for now.

**Comment by laurenceisla** [comment]
> Most importantly: Have you confirmed that you can actually use different logo files with both theme variants?

Ah, yeah, was just looking to see if that's possible at all. It seems to be by ditching `..image` and using raw html, or by using javascript and `..figure`, but it's not that straightforward for both cases.

If it can't be done, then adding a white background like Supabase or Neon is using right now would be a way out to keep them *kinda* consistent... anyhow, the sponsors need to approve them too.

> Is there any time-pressure in getting a dark theme out? Why not just wait until you have the logos?

Nope, was just my way to stop the PR from staying in limbo for too long, since I don't actually mind the inconsistency of the logos that much. 

In summary, this needs to be done:
- Check if it's feasible for the logos to change for each theme and ask the sponsors to provide dark mode logos
- Otherwise, ask if it's OK to sponsors to use white background for logos (maybe they provide one for both dark/light modes), this would be easier.

**Comment by laurenceisla** [comment]
> ... this could potentially work around the issue, if the landing page was always the same style and only the docs itself would be switched to dark?

Ah, I forgot about the landing page. Yes, that would be a good workaround actually.

> But I would not want to be stuck with it for technical reasons.

Now that I was checking around, there are some themes different from `rtd` that look good and give many out of the box solutions. For example, [Furo](https://github.com/pradyunsg/furo) seems to be a good alternative, it even has selecting image by dark/light theme and is compatible with all the extensions we use right now (except `sphinx-tabs`, but has an alternative) and looks easier to customize. The [documentation for PIP](https://pip.pypa.io/en/stable/) uses this theme.

The drawbacks are that there may be some issues that would come back like https://github.com/PostgREST/postgrest-docs/issues/351 and some features may be missing, like the breadcrumb.

Just to get the idea out there.

**Comment by laurenceisla** [comment]
> Would they work with readthedocs, though? 

Didn't check if it was straightforward enough, but I'm pretty sure it can be done since the [Pip docs](https://pip.pypa.io/en/stable/) have that button enabled.

**Comment by laurenceisla** [comment]
I installed the Furo theme to see how it looks like. The RTD button is activated without problems. It still needs some tweaks and the sidebar doesn't fully convince me, but the navigation using the right panel looks really good.

**Comment by laurenceisla** [comment]
> My gut feeling tells me that there would be more than just some "tweaks" needed here

True that, I noticed there are some other things that are not trivial to adapt. Well, at least we know that themes can be used if we need them in the future. For now, I'll go back and use the dark theme.

The good thing is there was an easy way to change images according to the theme. Basically hide/show the `` element using css `display` property. As an example, I used a dark mode image provided by Supabase in [their webpage](https://supabase.com/brand-assets). It's still pending to ask the other sponsors for their logos, but it can be done later.

Also added some background to the other transparent images to make them readable at least.

**Comment by laurenceisla** [comment]
OK, this should be officially completed. All the sponsors that did not provide a dark mode before were contacted. Most of them sent the dark mode and others confirmed that the logo is OK for both dark and light mode. Also updated the light mode logos with new ones for those who requested it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
  - `Prefer: params=single-object` is deprecated. Use [a function with a single unnamed JSON parameter](https://postgrest.org/en/latest/references/api/stored_procedures.html#s-proc-single-json) instead. - @steve-chavez
 
+### Documentation
```
Comment (laurenceisla) [inline CHANGELOG.md:30]: Decided to add a new subtitle to the CHANGELOG for changes in the docs.

File: nix/tools/docs.nix [hunk 2]
```diff
@@ -16,6 +16,7 @@ let
     ps.sphinx-tabs
     ps.sphinx-copybutton
     ps.sphinxext-opengraph
+    (ps.callPackage ../docs-extensions/sphinx-rtd-dark-mode.nix { })
```
[the remark this answers is not in the document]
Comment (laurenceisla) [reply nix/tools/docs.nix:19]: Cool, I copied the file as is.

**Comment by laurenceisla** [comment]
The `sphinx-rtd-dark-mode` is already merged to Nixpkgs as mentioned in https://github.com/PostgREST/postgrest/pull/3289#discussion_r1527186534. Will take a while to use that pkgs version, so I'm merging as it is right now.
